soh/soh/Enhancements/stairs.cpp

```diff
@@ -0,0 +1,675 @@
+#include "stairs.h"
+#include <string.h>
+
+struct ActorIdAllocTypePair {
+    u16 id;
+    u16 allocType;
+};
+
+#define DEFINE_ACTOR_INTERNAL(name, id, allocType) { id, allocType },
+#define DEFINE_ACTOR(name, id, allocType) { id, allocType },
+#define DEFINE_ACTOR_UNSET(_0)
+
+static const std::vector<ActorIdAllocTypePair> stairActorTable = {
+#include "tables/actor_table.h"
+};
+
+#undef DEFINE_ACTOR_INTERNAL
+#undef DEFINE_ACTOR_UNSET
+#undef DEFINE_ACTOR
+
+// Original size subtracted by various constant allocations which are always made on init, so those allocations don't need to be made there
+size_t stairsAllocSize = 0x1D4790 - 0x26960 - 0x2200 - (0x55 * 0x60) - (3 * (48 * 16) / 2) - (4 * (32 * 32 * 4)) - 0x1000 - (20 * sizeof(MtxF));
+Arena sStairsArena;
+
+// using sizes from the debug rom
+std::unordered_map<u16, size_t> actorOverlaySizes = {
```

Comment:
To the best of my understanding, this should probably be static const. So `static const std::unordered_map...`. Assuming you aren't changing this at runtime any. Shouldn't impact functionality, more of a correctness factor for storage/type specification.